app/page.tsx

```diff
@@ -37,34 +37,36 @@ type Entry = {
   lexicon: string;
 };
 
+const predProblemsA: ProbCol = predProblems as ProbCol as const;
+const propProblemsA: ProbCol = propProblems as ProbCol as const;
+
 export default function Home() {
-  const predProblemsA: ProbCol = predProblems as ProbCol;
-  const propProblemsA: ProbCol = propProblems as ProbCol;
+  // user interaction
   const [logic, setLogic] = useState("prop");
-  const [isSyntaxVisible, setIsSyntaxVisible] = useState(false);
   const [problemCollection, setProblemCollection] = useState<ProbCol>(
     propProblemsA as ProbCol
   );
   const [selectedProblem, setSelectedProblem] = useState(1);
   const [userFormula, setUserFormula] = useState("");
   const [userSoa, setUserSoa] = useState([{ symbol: "", lexicon: "" }]);
-  const [error, setError] = useState(false);
-  const [errorText, setErrorText] = useState("");
-  const [success, setSuccess] = useState(false);
-  const [successText, setSuccessText] = useState("");
-  const [note, setNote] = useState(false);
-  const [noteText, setNoteText] = useState("");
```

Comment:
These `...Text` were extra sources of truth. I prefer using as few of these as possible so that it is harder to get your code into a bad spot (where you set one but not the other).